src/bridge/MatrixHandler.ts

```diff
@@ -1046,13 +1052,55 @@ export class MatrixHandler {
-            const eventId = event.content["m.relates_to"]["m.in_reply_to"].event_id;
-            const reply = await this.textForReplyEvent(event, eventId, ircRoom);
-            if (reply !== null) {
-                ircAction.text = reply.formatted;
-                cacheBody = reply.reply;
+        // special handling for replies and threads
+        if (event.content["m.relates_to"]) {
+            if (event.content["m.relates_to"]["m.in_reply_to"]) {
+                const eventId = event.content["m.relates_to"]["m.in_reply_to"].event_id;
+                const reply = await this.textForReplyEvent(event, eventId, ircRoom);
+                if (reply !== null) {
+                    ircAction.text = reply.formatted;
+                    cacheBody = reply.reply;
+                }
+            } else if (event.content["m.relates_to"]?.rel_type === "io.element.thread") {
+                const threadId = event.content["m.relates_to"].event_id;
+                let lastReplyId = this.lastMessageInThreadCache.get(threadId);
+                if (!lastReplyId) {
+                    try {
+                        const getThread = async (from?: string) => this.ircBridge.getAppServiceBridge().getIntent().matrixClient.doRequest(
+                            'GET', `/_matrix/client/unstable/rooms/${event.room_id}/relations/${threadId}` + (from ? `?from=${from}`: '')
+                        );
+                        let from: string|undefined;
+                        // locate the message we're now formatting in the thread chain and find its predecessor
+                        for (let i = 0; i < this.config.threadLookupRequestLimit; i++) {
+                            let thread = await getThread(from);
+
+                            const currentMessageIdx = thread['chunk'].findIndex((ev: any) => ev.event_id = event.event_id);
+                            if (currentMessageIdx == -1) {
+                                from = thread.next_batch;
+                                if (!from) break;
+                                continue;
+                            } else {
+                                if (currentMessageIdx === thread['chunk'].length) {
+                                    // we're at a chunk boundary, need to load one more
+                                    thread = await getThread(thread.next_batch);
+                                    lastReplyId = thread['chunk'][0].event_id;
+                                } else {
+                                    lastReplyId = thread['chunk'][currentMessageIdx + 1].event_id;
+                                }
+                                break;
+                            }
+                        }
+                    } catch (err) {
+                        console.error('Error fetching thread from homeserver:', err);
+                    }
+                    lastReplyId ||= threadId; // fallback to the original message if we really have no better idea
```

Comment:
Personal feeling is this logic feels too complex to sit in this function. Should at least be it's own function, if not in the bot/bridge sdk itself.